provider/src/main/scala/com/actian/spark_vector/provider/RequestHandler.scala

```diff
@@ -92,17 +93,21 @@ class RequestHandler(sqlContext: SQLContext, val auth: ProviderAuth) extends Log
     writeJobResult(result)
   }
 
+  private def getMessage(cause: Throwable): String = cause.getMessage + (if (cause.getCause != null) s", caused by: ${getMessage(cause.getCause)}" else "")
```

Comment:
Based on [1], how about we use: Iterator.iterate(cause)(_.getCause).takeWhile(_ != null).mkString(", caused by: ") ? (haven't check with `Throwable`s but for list of strings, ints it works)

It looks nicer, is not that the stack may explode if the exception has a deep cause root..which I don't think is our case here.

Note that cause.toString, which will be used in building the string (no concat btw, given mkString uses StringBuilder) is going to have <class>:<message> format, hence the message is going to look like "<class>:<message>, caused by: <class>:<message> ..."

[1] http://blog.bruchez.name/2012/07/scala-iterators-and-iteratoriterate.html